forge/src/main/java/me/shedaniel/architectury/registry/forge/TradeRegistryImpl.java

```diff
@@ -0,0 +1,65 @@
+package me.shedaniel.architectury.registry.forge;
+
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMaps;
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import me.shedaniel.architectury.forge.ArchitecturyForge;
+import net.minecraft.world.entity.npc.VillagerProfession;
+import net.minecraft.world.entity.npc.VillagerTrades;
+import net.minecraft.world.entity.npc.WanderingTrader;
+import net.minecraftforge.event.village.VillagerTradesEvent;
+import net.minecraftforge.event.village.WandererTradesEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.util.*;
+
+@Mod.EventBusSubscriber(modid = ArchitecturyForge.MOD_ID)
+public class TradeRegistryImpl {
+    
+    private static final Map<VillagerProfession, Int2ObjectMap<List<VillagerTrades.ItemListing>>> TRADES_TO_ADD = new HashMap<>();
+    private static final List<VillagerTrades.ItemListing> WANDERER_TRADER_TRADES_GENERIC = new ArrayList<>();
+    private static final List<VillagerTrades.ItemListing> WANDERER_TRADER_TRADES_RARE = new ArrayList<>();
+    
+    public static void registerVillagerTrade(VillagerProfession profession, int level, VillagerTrades.ItemListing... trades) {
+        if(level < 1 || level > 5){
+            throw new RuntimeException("Villager Trade level has to be between 1 and 5!");
```

Comment:
IllegalArgumentException, and print the current invalid value (Change this in the fabric one as well)

Reply:
done